Framework/NexusGeometry/src/NexusGeometrySave.cpp

```diff
@@ -278,8 +283,22 @@ inline void writePixelData(H5::Group &grp,
   posy.reserve(nPixels);
   posz.reserve(nPixels);
 
-  // only need to write pixel shape group once in this case
-  if (shapesAreValidAndHomogeneous) {
+  if (!cylindersExist) {
```

Comment:
can we do this check before we start reserving memory above?

Reply:
The xyz positions are only related to the writing of the pixel offsets from the bank, even if they dont have shapes. If nPixels is ever empty, on line 235, the function will exit. I will add comments